streaming/java/streaming-runtime/src/main/java/org/ray/streaming/runtime/core/master/resourcemanager/ResourceManagerImpl.java

```diff
@@ -0,0 +1,200 @@
+import org.ray.streaming.runtime.master.JobRuntimeContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ResourceManagerImpl implements ResourceManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ResourceManagerImpl.class);
+
+  //Container used tag
+  private static final String CONTAINER_ENGAGED_KEY = "CONTAINER_ENGAGED_KEY";
+
+  /**
+   * Job runtime context.
+   */
+  private JobRuntimeContext runtimeContext;
+
+  /**
+   * Resource related configuration.
+   */
+  private ResourceConfig resourceConfig;
+
+  /**
+   * Slot assign strategy.
+   */
+  private SlotAssignStrategy slotAssignStrategy;
+
+  /**
+   * Resource description information.
+   */
+  private final Resources resources;
+
+  private final ScheduledExecutorService scheduledExecutorService;
+
+  public ResourceManagerImpl(JobRuntimeContext runtimeContext) {
+    this.runtimeContext = runtimeContext;
+    StreamingMasterConfig masterConfig = runtimeContext.getConf().masterConfig;
+
+    this.resourceConfig = masterConfig.resourceConfig;
+    this.resources = new Resources(resourceConfig);
+    LOG.info("ResourceManagerImpl begin init, conf is {}, resources are {}.",
+        resourceConfig, resources);
+
+    SlotAssignStrategyType slotAssignStrategyType = SlotAssignStrategyType.PIPELINE_FIRST_STRATEGY;
+
+    this.slotAssignStrategy = SlotAssignStrategyFactory.getStrategy(slotAssignStrategyType);
+    this.slotAssignStrategy.setResources(resources);
+    LOG.info("Slot assign strategy: {}.", slotAssignStrategy.getName());
+
+    this.scheduledExecutorService = Executors.newScheduledThreadPool(1);
+    long intervalSecond = resourceConfig.resourceCheckIntervalSecond();
+    this.scheduledExecutorService.scheduleAtFixedRate(
+        Ray.wrapRunnable(this::checkAndUpdateResources), 0, intervalSecond, TimeUnit.SECONDS);
+
+    LOG.info("ResourceManagerImpl init success.");
+  }
+
+  @Override
+  public Map<String, Double> allocateResource(final Container container,
+      final Map<String, Double> requireResource) {
+    LOG.info("Start to allocate resource for actor with container: {}.", container);
+
+    // allocate resource to actor
+    Map<String, Double> resources = new HashMap<>();
+    Map<String, Double> containResource = container.getAvailableResource();
+    for (Map.Entry<String, Double> entry : containResource.entrySet()) {
+      if (requireResource.containsKey(entry.getKey())) {
+        double availableResource = entry.getValue() - requireResource.get(entry.getKey());
+        entry.setValue(availableResource);
+        resources.put(entry.getKey(), requireResource.get(entry.getKey()));
+      }
+    }
+
+    LOG.info("Allocate resource: {} to container {}.", requireResource, container);
+    return resources;
+  }
+
+  @Override
+  public void deallocateResource(final Container container,
+      final Map<String, Double> releaseResource) {
+    LOG.info("Start deallocate resource for container {}.", container);
```

Comment:
```suggestion
    LOG.info("Deallocating resource for container {}.", container);
```
Use `doing` to log something that is happening, and `did` for something already done